cmdstanpy/compiler_opts.py

```diff
@@ -165,6 +170,10 @@ def validate_stanc_opts(self) -> None:
                     self._cpp_options = {'STAN_OPENCL': 'TRUE'}
                 else:
                     self._cpp_options['STAN_OPENCL'] = 'TRUE'
+            elif key == 'O':
```

Comment:
I don't think we want to pass these as key=val. `--O` is still valid all by itself for backwards compatibility (synonymous with `--Oexperimental`)

I think that the best way to implement this is just to add `O0`, `O1`, and `Oexperimental` to the list of allowed keys. If we wanted to get really fancy we could try to enforce that at most one of these is used, but even that isn't strictly necessary (the compiler will just use the final one as the definitive argument)

Reply:
Ah, rather having (key, val ) pairs. Yes, that would make more sense for a user. I can change that.

Reply:
that should fix it